adapta/storage/distributed_object_store/datastax_astra/astra_client.py

```diff
@@ -414,6 +415,16 @@ def map_to_column(  # pylint: disable=R0911
             ):  # assume all enums are strings - for now
                 return (columns.Text,)
             if get_origin(python_type) == list:
+                args = typing.get_args(python_type)
+                if get_origin(args[0]) == dict:
+                    dict_args = typing.get_args(args[0])
+                    return (
+                        columns.List,
+                        columns.Map(
+                            map_to_column(dict_args[0])[0],
+                            map_to_column(dict_args[1])[0],
```

Comment:
Beware this assumes that the dict type can only have simple keys and values.

Structures like `list[dict[str,dict[str,str]]]` will not work here. Actually, same goes for the other list and map types in this function.

If you do not fix this, please open a new issue for this

Reply:
Made this issue https://github.com/SneaksAndData/adapta/issues/425